src/helper/selection-tools/bounding-box-tool.js

```diff
@@ -105,27 +92,92 @@ class BoundingBoxTool {
+    }
+    _updateCursor (event, hitOptions) {
+        const { mode, hitResult } = this._determineMode(event, false, hitOptions);
+        if (hitResult) {
+            if (mode === BoundingBoxModes.MOVE) {
+                this.setCursor('grab');
+            } else if (mode === BoundingBoxModes.ROTATE) {
+                this.setCursor('grab');
+            } else if (mode === BoundingBoxModes.SCALE) {
+                this.setSelectionBounds();
+                if (this.boundsPath.position.x === hitResult.item.position.x) {
+                    this.setCursor('ns-resize');
+                } else if (this.boundsPath.position.y === hitResult.item.position.y) {
+                    this.setCursor('ew-resize');
+                } else if (this.boundsPath.position.y < hitResult.item.position.y) {
+                    if (this.boundsPath.position.x > hitResult.item.position.x) {
+                        this.setCursor('nesw-resize');
+                    } else {
+                        this.setCursor('nwse-resize');
+                    }
+                } else {
+                    if (this.boundsPath.position.x > hitResult.item.position.x) {
+                        this.setCursor('nwse-resize');
+                    } else {
+                        this.setCursor('nesw-resize');
+                    }
+                }
+            }
+        } else {
+            this.setCursor('default');
+        }
+    }
+    _determineMode (event, multiselect, hitOptions) {
+        const hitResults = paper.project.hitTestAll(event.point, hitOptions);
+
+        let mode;
+
+        // Prefer scale to trigger over rotate, and scale and rotate to trigger over other hits
+        let hitResult = hitResults[0];
+        for (let i = 0; i < hitResults.length; i++) {
+            if (hitResults[i].item.data && hitResults[i].item.data.isScaleHandle) {
+                hitResult = hitResults[i];
+                mode = BoundingBoxModes.SCALE;
+                break;
+            } else if (hitResults[i].item.data && hitResults[i].item.data.isRotHandle) {
+                hitResult = hitResults[i];
+                mode = BoundingBoxModes.ROTATE;
+            }
+        }
+        if (!mode) {
+            mode = BoundingBoxModes.MOVE;
+        }
+
+        return {mode, hitResults, hitResult};
+    }
     onMouseDrag (event) {
         if (event.event.button > 0 || !this.mode) return; // only first mouse button
         this._modeMap[this.mode].onMouseDrag(event);
     }
-    onMouseUp (event) {
+    onMouseUp (event, hitOptions) {
         if (event.event.button > 0 || !this.mode) return; // only first mouse button
         this._modeMap[this.mode].onMouseUp(event);
 
+        // Update the cursor to match the handle again. This has to be done a tick later because, if called
+        // immediately, the handles won't be visible yet (they are hidden while dragging).
+        setTimeout(() => {
+            this._updateCursor(event, hitOptions);
```

Comment:
Yeah, I see. I think it would be better to take the latter option. My (long, unwieldy) paragraph above is somewhat relevant here:

> Good point, thanks. The problem I faced with this is that I need to do `handleMouseMove` on those tools as well, i.e. I need to add [this](https://github.com/LLK/scratch-paint/pull/762/files#diff-a4da56a05d313254413ded8e7d51b386R144) to the other tools. But those tools don't have access to `getHitOptions`. I'll probably have to go with a solution like you mentioned [here](https://github.com/LLK/scratch-paint/pull/762/#discussion_r234003522) (passing that function around). Input on how exactly that should be done, i.e. where that `getHitOptions` function should be stored (so we aren't duplicating code everywhere)? ATM I think we just define the `method(val) { dispatch(method(val)) }`-style thing for reducers in every file that uses those reducers, but obviously that's not an option for this. Is there a util function somewhere for this type of function? (`getHitOptions` works well as a separate function, BTW, since it doesn't need to be tied to any class - it doesn't ever reference `this`.)

NB - we do have a `getHitOptions` function in every relevant class (I think), i.e. the ones make use of a `boundingBoxTool`; but it feels like this function is getting duplicated around a lot. Not that it's particularly necessary to do this PR (I'm pretty sure, anyway), but could there be a better place to store that function, i.e. a utility function, considering the quoted paragraph above?

Reply:
[Resolved](https://github.com/LLK/scratch-paint/pull/762#issuecomment-439273560) :)